src/main/java/picard/sam/SplitSamByNumberOfReads.java

```diff
@@ -106,11 +110,17 @@ protected int doWork() {
         final ProgressLogger firstPassProgress = new ProgressLogger(log, 1000000, "Counted");
         if (TOTAL_READS_IN_INPUT == 0) {
             final SamReader firstPass = readerFactory.referenceSequence(REFERENCE_SEQUENCE).open(INPUT);
+            final SAMRecordIterator firstPassIterator = firstPass.iterator();
+            if (readerIterator.hasNext() && !firstPassIterator.hasNext()) {
```

Comment:
hmmm. according to the javadoc here: htsjdk.samtools.BAMFileReader#getIterator() only one iterator is allowed to exist at a given time, so I'm not sure how this is working....I think that to do it properly, you need to open another reader to the same file.